com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDGpuLightsBuilder.LightLoop.cs

```diff
@@ -0,0 +1,483 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace UnityEngine.Rendering.HighDefinition
+{
+    internal partial class HDGpuLightsBuilder
+    {
+        #region internal HDRP API
+
+        //Preallocates number of lights for bounds arrays and resets all internal counters. Must be called once per frame per view always.
+        public void NewFrame(HDCamera hdCamera, int maxLightCount)
+        {
+            int viewCounts = hdCamera.viewCount;
+            if (viewCounts > m_LighsPerViewCapacity)
+            {
+                m_LighsPerViewCapacity = viewCounts;
+                m_LightsPerView.ResizeArray(m_LighsPerViewCapacity);
+            }
+
+            m_LightsPerViewCount = viewCounts;
+
+            int totalBoundsCount = maxLightCount * viewCounts;
+            int requestedBoundsCount = Math.Max(totalBoundsCount, 1);
+            if (requestedBoundsCount > m_LightBoundsCapacity)
+            {
+                m_LightBoundsCapacity = Math.Max(Math.Max(m_LightBoundsCapacity * 2, requestedBoundsCount), ArrayCapacity);
```

Comment:
Is the *2 here for the two eyes? I think we in theory support arbitrary view counts for XR